src/sql/parts/notebook/notebook.component.ts

```diff
@@ -331,6 +337,47 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		}
 	}
 
+	// We can keep this method in common utilities
+	private getFilePathFromRecentWorkspace(untitledResource: URI): string {
```

Comment:
I don't understand what function is doing - can you rename to state its purpose? And/or add a doc comment? I _think_ from reading it this is trying to get a sample filename for use when prompting users to save?

Reply:
This is to get the recently used workspace in local computer. It is queried against HistoryService to pull the info. Untitled notebook doesn't have local path info in order to save file by prompting.